cmake/lint.cmake

```diff
@@ -0,0 +1,34 @@
+function(LINT)
+  cmake_parse_arguments(LINT "QUIET" "PROGRAM" "FLAGS;FILES" ${ARGN})
+
+  if(LINT_QUIET)
+    set(OUTPUT_QUIET OUTPUT_QUIET)
+  elseif()
+    set(OUTPUT_QUIET "")
+  endif()
+
+  find_program(PROGRAM_EXISTS ${LINT_PROGRAM})
+  if(PROGRAM_EXISTS)
+    execute_process(COMMAND ${LINT_PROGRAM} ${LINT_FLAGS} ${LINT_FILES}
```

Comment:
Does this always run, even if there are no changes since the last run? If so, that's a regression. We can think about that in a followup PR though.